optuna/storages/_cached_storage.py

```diff
@@ -96,13 +96,17 @@ def __setstate__(self, state: Dict[Any, Any]) -> None:
         self.__dict__.update(state)
         self._lock = threading.Lock()
 
-    def create_new_study(self, study_name: Optional[str] = None) -> int:
+    def create_new_study(
+        self, directions: Sequence[StudyDirection], study_name: Optional[str] = None
+    ) -> int:
 
-        study_id = self._backend.create_new_study(study_name)
+        study_id = self._backend.create_new_study(directions=directions, study_name=study_name)
```

Comment:
How about adding the following test case in `test_with_server.py`?

```diff
$ git diff
diff --git a/tests/storages_tests/test_with_server.py b/tests/storages_tests/test_with_server.py
index 18b4ae481..fecb2fd31 100644
--- a/tests/storages_tests/test_with_server.py
+++ b/tests/storages_tests/test_with_server.py
@@ -1,4 +1,4 @@
-from concurrent.futures import ProcessPoolExecutor
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
 import os
 import pickle
 from typing import Sequence
@@ -148,7 +148,15 @@ def test_store_nan_intermediate_values(storage: BaseStorage) -> None:
     assert np.isnan(got_value)


-def test_multiprocess(storage: BaseStorage) -> None:
+def test_multithread_create_study(storage: BaseStorage) -> None:
+    with ThreadPoolExecutor(10) as pool:
+        for _ in range(10):
+            pool.submit(optuna.create_study, storage=storage, study_name="test-multithread-create-study", load_if_exists=True)
+
+    assert len(storage.get_all_studies()) == 1
+
+
+def test_multiprocess_run_optimize(storage: BaseStorage) -> None:
     n_workers = 8
     n_trials = 20
     study_name = _STUDY_NAME
```

Reply:
Sounds good! I will add this test.

Reply:
This test [failed](https://github.com/optuna/optuna/actions/runs/3704495678/jobs/6277185743#step:9:33)...